File: Day28/Solution.rs
use std::io;
use regex::Regex;

fn main() {
    let pattern = r".+@gmail\.com";
    let re = Regex::new(pattern).unwrap();
    let mut input = String::new();
    io::stdin().read_line(&mut input).expect("Not a valid string");
    let n = input.trim_end().parse::<u32>().expect("Not a valid integer");
    for _ in 0..n {
        io::stdin().read_line(&mut input).expect("Not a valid string");
    }
    let mut result = input.lines()
                          .into_iter()
                          .filter(|line| re.is_match(&line))
                          .filter_map(|line| line.split_whitespace().next())
                          .collect::<Vec<&str>>();
    result.sort();
    result.iter().for_each(|name| println!("{}", name));
}

File: Day28/Solution.swift
import Foundation

extension NSRegularExpression {
    func matches(_ string: String) -> Bool {
        let range = NSRange(location: 0, length: string.utf16.count)
        return firstMatch(in: string, options: [], range: range) != nil
    }
}

let testCases = Int(readLine() ?? "") ?? 0
let regex = try? NSRegularExpression(pattern: "[a-z]@gmail.com")
var gmailUsers = [String]()

for _ in 0..<testCases {
    let input = readLine()?.split(separator: " ").map {String($0)}
    if regex?.matches(input?[1] ?? "") ?? false {
        gmailUsers.append(input?[0] ?? "")
    }
}

gmailUsers.sort()
gmailUsers.forEach{print($0)}
